parachains/runtimes/assets/statemine/src/weights/xcm/mod.rs

```diff
@@ -35,7 +35,12 @@ impl WeighMultiAssets for MultiAssetFilter {
 		match self {
 			Self::Definite(assets) =>
 				weight.saturating_mul(assets.inner().into_iter().count() as u64),
-			Self::Wild(_) => weight.saturating_mul(MAX_ASSETS as u64),
+			Self::Wild(asset) => match asset {
+				All => weight.saturating_mul(MAX_ASSETS as u64),
+				AllOf { .. } => weight,
```

Comment:
I'm actually still uncertain about this -- comments on AllOfCounted says that each individual instance of an NFT is considered a different class of asset, and so wouldn't it make more sense to use the number of NFT instances as an upper bound here?

Reply:
Hmm, good question. Let me check it out and come back to you.

Reply:
Yeah, I think this makes sense. I'm not sure though where the NFT instance number should be coming from in this case.

Reply:
So this requires you to actually look inside the holding register and determine just how many `AssetInstance`s there are... We could match against `fun` in `AllOf` and see if it's a `NonFungible`, and if so, it would basically be `weight * MaxAssetsIntoHolding::get()`, otherwise it's just `weight`.

Reply:
Thanks!

Reply:
@KiChjang One thing that disturbs me a bit is the comment below, should we then do something like `weight * MaxAssetsIntoHolding::get() * 2` to be safe? 

```rust
	/// The maximum number of assets we target to have in the Holding Register at any one time.
	///
	/// NOTE: In the worse case, the Holding Register may contain up to twice as many assets as this
	/// and any benchmarks should take that into account.
	type MaxAssetsIntoHolding: Get<u32>;
```